src/PostageStamp.sol

```diff
@@ -382,43 +392,22 @@ contract PostageStamp is AccessControl, Pausable {
         pot += validChunkCount * (lastExpiryBalance - leb);
     }
 
-    /**
-     * @notice Reclaims a limited number of expired batches
-     * @dev Can be used if reclaiming all expired batches would exceed the block gas limit, causing other
-     * contract method calls to fail.
-     * @param limit The maximum number of batches to expire.
-     */
-    function expireLimited(uint256 limit) external {
-        uint256 i;
-        for (i = 0; i < limit; i++) {
-            if (empty()) break;
-            bytes32 fbi = firstBatchId();
-            if (remainingBalance(fbi) > 0) break;
-            Batch storage batch = batches[fbi];
-            uint256 batchSize = 1 << batch.depth;
-            validChunkCount -= batchSize;
-            pot += batchSize * (batch.normalisedBalance - lastExpiryBalance);
-            tree.remove(fbi, batch.normalisedBalance);
-            delete batches[fbi];
-        }
-    }
 
     /**
-     * @notice Inidicates there are batches waiting to be processed by _expire_.
+     * @notice Indicates whether expired batches exist.
      */
-    function hasExpired() public view returns(bool) {
-        bytes32 fbi = firstBatchId();
-        if (remainingBalance(fbi) > 0) {
+    function expiredBatchesExist() public view returns (bool) {
+        if (empty()){
             return false;
         }
-        return true;
+        return !(remainingBalance(firstBatchId()) > 0);
```

Comment:
this is funky :)

just do `return remainingBalance(firstBatchId()) < 0`

Reply:
! > 0 is not the same as < 0, but actually <= 0, if zero matters.